verilog/preprocessor/verilog_preprocess_test.cc

```diff
@@ -791,5 +791,25 @@ endmodule
   }
 }
 
+TEST(VerilogPreprocessTest, SetExternalDefines) {
+  // Test case input tokens.
+  const verible::TokenSequence test_case_tokens = {
+      {verible::TokenInfo(MacroIdentifier, "`MACRO1")},
+      {MacroIdentifier, "`MACRO2"}};
+  verible::TokenStreamView test_case_stream_view;
+  verible::InitTokenStreamView(test_case_tokens, &test_case_stream_view);
+
+  VerilogPreprocess::Config test_config({.expand_macros = true});
+  VerilogPreprocess preprocessor(test_config);
+
+  preprocessor.SetExternalDefine("MACRO1", "VALUE1");
+  preprocessor.SetExternalDefine("MACRO2", "VALUE2");
+
+  const auto& pp_data = preprocessor.ScanStream(test_case_stream_view);
+
+  EXPECT_THAT(pp_data.preprocessed_token_stream[0]->text(), "VALUE1");
+  EXPECT_THAT(pp_data.preprocessed_token_stream[1]->text(), "VALUE2");
+}
```

Comment:
I couldn't use the testing class in here, because it doesn't use `SetExternalDefine()`, I also didn't want to directly use the lexer, as that would need some extra includes, and would be the same way the preprocessor tool is used.

This test also uses expanding macros feature.

Reply:
So ideally weProbably in one of the next steps in this journey, we need to tell `VerilogAnalyzer` about `FileList`, so that we can inject the necessary information that way.
Make sure to add a TODO here, so that we don't forget to later run everything through `PreprocessorTester`

Reply:
Once we also have ``undef implemented, we can also see if undef properly undefines the value set by SetExternalDefine.